lib/puppet/file_system/jruby.rb

```diff
@@ -0,0 +1,23 @@
+require 'puppet/file_system/posix'
+
+class Puppet::FileSystem::JRuby < Puppet::FileSystem::Posix
+  def unlink(*paths)
```

Comment:
It's not clear to me what unlink is doing in this PR. I don't see anything in the comments or commit titles.

Reply:
Would need @joshcooper to comment on this as this is a commit that's already in 6.

Reply:
IIRC `close!` calls `unlink` and the newly added tests failed on JRuby due to something with the parent directory. I should have mentioned that in the commit message.